linksharing/handler.go

```diff
@@ -101,20 +94,15 @@ func (handler *Handler) serveHTTP(w http.ResponseWriter, r *http.Request) (err e
 		}
 	}()
 
-	b, err := p.OpenBucket(ctx, bucket, access.EncryptionAccess)
-	if err != nil {
-		handler.handleUplinkErr(w, "open bucket", err)
+	b, err := p.StatBucket(ctx, bucket)
```

Comment:
why do we need this StatBucket call?

Reply:
We need the stat bucket call to return an error if the bucket does not exist. Else we would just return object not exist.

Reply:
yeah returning object not exist is totally acceptable, considering the StatBucket call adds a couple hundred ms to the overall operation. getting rid of this for performance for the normal case is worth it